src/converter/Main.java

```
@@ -68,10 +92,88 @@ public static void convert(String input, String output, Arguments args) throws I
         }
     }
 
+    private static void processDifferential(String file1, String file2, String output, Arguments args)
+            throws IOException {
+        System.out.print("Creating differential flame graph: " + getFileName(file1) + " vs " + getFileName(file2)
+                + " -> " + getFileName(output) + " ");
+        System.out.flush();
+
+        long startTime = System.nanoTime();
+
+        try {
+            validateFileFormats(file1, file2);
+
+            ProfileData profile1 = getProfileData(file1, args);
+            ProfileData profile2 = getProfileData(file2, args);
+            DifferentialResult differentialResult = DiffFolded.process(profile1, profile2);
+
+            createDifferentialFlameGraph(differentialResult, output, args);
+        } catch (IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new IOException("Failed to create differential: " + e.getMessage(), e);
+        }
+
+        long endTime = System.nanoTime();
+        System.out.print("# " + (endTime - startTime) / 1000000 / 1000.0 + " s\n");
+    }
+
+    private static void validateFileFormats(String file1, String file2) throws IOException {
+        String format1 = getFileFormat(file1);
+        String format2 = getFileFormat(file2);
+
+        if (!format1.equals(format2)) {
+            System.err.println("Error: Cannot create differential flame graph with different file formats");
+            System.err.println("File 1: " + getFileName(file1) + " (" + format1 + ")");
+            System.err.println("File 2: " + getFileName(file2) + " (" + format2 + ")");
+            System.err.println("Both files must be the same format (JFR, HTML, or collapsed)");
```

otlp not getting any love?

I can see that `otlp` to `collapsed` is not yet supported
Let's throw an exception at least?